FlowCrypt/Functionality/Mail Provider/Contacts Provider/CloudContactsProvider.swift

```diff
@@ -6,64 +6,70 @@
 //  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
 //
 
-import Foundation
+import FlowCryptCommon
 import Promises
+import GoogleAPIClientForREST_PeopleService
 
 protocol CloudContactsProvider {
     func searchContacts(query: String) -> Promise<[String]>
 }
 
-final class UserContactsProvider {
-    private enum Constants {
-        static let scheme = "https"
-        static let host = "www.google.com"
-        static let searchPath = "/m8/feeds/contacts/default/thin"
-    }
+enum CloudContactsProviderError: Error {
+    /// People API response parsing
+    case failedToParseData(Any?)
+    /// Provider Error
+    case providerError(Error)
+}
 
-    private let dataService: DataService
+final class UserContactsProvider {
+    private let logger = Logger.nested("UserContactsProvider")
+    private let userService: GoogleUserServiceType
+    private var peopleService: GTLRPeopleServiceService {
+        let service = GTLRPeopleServiceService()
 
-    private var token: String? {
-        dataService.token
-    }
+        if userService.authorization == nil {
+            logger.logWarning("authorization for current user is nil")
+        }
 
-    init(dataService: DataService = .shared) {
-        self.dataService = dataService
+        service.authorizer = userService.authorization
+        return service
     }
 
-    private func components(for path: String) -> URLComponents {
-        var components = URLComponents()
-        components.scheme = Constants.scheme
-        components.host = Constants.host
-        components.path = path
-        return components
+    init(userService: GoogleUserServiceType = GoogleUserService()) {
+        self.userService = userService
+        
+        // Warmup query for contacts cache
+        _ = self.searchContacts(query: "")
```

Comment:
Interesting - is this for a google servers cache or internal cache? When will this trigger, when loading the compose view?

Reply:
It's for warming up google lazy cache, documentation mentions this request here https://developers.google.com/people/v1/contacts#search_the_users_contacts

App sends it when compose screen is loading.